### PR TITLE
For #82: Introduce deletion of errors.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #82: Introduce deletion of errors.
- added `TkErrorDelete` endpoint for error deletion;
- wired `TkErrorDelete` in `TkApp`;
- corrected `TkErrors.source` method;
- corrected `DyErrorITCase` / `DyErrorsITCase` tests

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#84](https://github.com/yegor256/wring/issues/84) is now in scope, role is `REV`

**Comment by paulodamaso** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/wring/issues/84#issuecomment-435030472))

@paulodamaso This is what I know about this job in [C7FCB1EQN](https://www.0crat.com/p/C7FCB1EQN), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#84](https://github.com/yegor256/wring/issues/84) is [in scope](http://www.zerocracy.com/policy.html#14) for 2days
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @paulodamaso[/z](https://www.0crat.com/u/paulodamaso): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C7FCB1EQN?q=%7Bjob%3A%27gh%3Ayegor256%2Fwring%2384%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by carlosmiranda** [comment]
@0crat assign @pinaf

**Comment by 0crat** [comment]
> @0crat assign @pinaf ([here](https://github.com/yegor256/wring/issues/84#issuecomment-440497114))

@carlosmiranda @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) not enough funds available in the project, can't set budget of job [#84](https://github.com/yegor256/wring/issues/84), see [§21](http://www.zerocracy.com/policy.html#21); @pinaf[/z](https://www.0crat.com/u/pinaf) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by paulodamaso** [comment]
@carlosmiranda Would you kindly review and merge this one? This task is preventing me from receiving another tasks due to a full agenda, and I have submitted it almost a month ago.

**Comment by paulodamaso** [comment]
@carlosmiranda ping

File: src/test/java/io/wring/dynamo/DyErrorITCase.java [hunk 1]
```diff
@@ -63,4 +66,21 @@ public void asXembly() throws Exception {
...
+    @Test
+    public void deletesErrors() throws Exception {
+        final User user = new DyUser(new Dynamo(), "boris");
+        final Errors errors = user.errors();
+        errors.register("error", "message");
+        final Error error = errors.iterate().iterator().next();
+        error.delete();
+        MatcherAssert.assertThat(
+            new LengthOf(errors.iterate()),
+            new IsEqual<>(0)
```
Comment (carlosmiranda) [inline src/test/java/io/wring/dynamo/DyErrorITCase.java]: @paulodamaso why not use `IsEmptyIterable`?

Comment (paulodamaso) [reply src/test/java/io/wring/dynamo/DyErrorITCase.java]: @carlosmiranda oops

**Comment by paulodamaso** [comment]
@carlosmiranda Done, please take a look

**Comment by carlosmiranda** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#84](https://github.com/yegor256/wring/issues/84) is not in the agenda of @pinaf[/z](https://www.0crat.com/u/pinaf), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job [#84](https://github.com/yegor256/wring/issues/84) is not in the agenda of @pinaf[/z](https://www.0crat.com/u/pinaf), can't inspect

**Comment by 0crat** [comment]
Code review was too long (20 days), architects (@carlosmiranda) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#84](https://github.com/yegor256/wring/issues/84) is now out of scope

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@041da5f8-f9e0-45ec-8176-7f0140d878bb, thread: AsyncProc-4
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.cost.funding.send_zold
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: groovy.lang.MissingMethodException: No signature of method: com.zerocracy.pmo.banks.Zold.pay() is applicable for argument types: (java.lang.String, com.zerocracy.cash.Cash$S, java.lang.String) values: [carlosmiranda, $5.00, none]
Possible solutions: pay(java.lang.String, com.zerocracy.cash.Cash, java.lang.String, java.lang.String), any(), wait(), find(), dump(), grep()
org.codehaus.groovy.runtime.ScriptBytecodeAdapter[71] groovy.lang.MissingMethodException: No signature of method: com.zerocracy.pmo.banks.Zold.pay() is applicable for argument types: (java.lang.String, com.zerocracy.cash.Cash$S, java.lang.String) values: [carlosmiranda, $5.00, none]
Possible solutions: pay(java.lang.String, com.zerocracy.cash.Cash, java.lang.String, java.lang.String), any(), wait(), find(), dump(), grep()
```

[0.39.6](https://github.com/zerocracy/farm): CID: [726ac397-e362-4f6a-beb9-4878cf41cb55](https://www.0crat.com/footprint/C7FCB1EQN/726ac397-e362-4f6a-beb9-4878cf41cb55), Type: "Send zold"

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/wring/issues/84#issuecomment-447562859))

@ypshenychka This is what I know about this job in [C7FCB1EQN](https://www.0crat.com/p/C7FCB1EQN), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#84](https://github.com/yegor256/wring/issues/84) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 3days
  * @pinaf[/z](https://www.0crat.com/u/pinaf) will receive 30 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C7FCB1EQN?q=%7Bjob%3A%27gh%3Ayegor256%2Fwring%2384%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by carlosmiranda** [comment]
@ypshenychka this wasn't assigned to any reviewer for funding reasons, so quality should be "bad"

**Comment by ypshenychka** [comment]
@carlosmiranda got it, thanks

**Comment by ypshenychka** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
> @0crat quality bad ([here](https://github.com/yegor256/wring/issues/84#issuecomment-447622662))

@ypshenychka The project doesn't have enough funds, can't make a payment

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**